### DISCUSSION TITLE
How to use useData in Layout

### DISCUSSION THREAD

**Discussion by ierehon1905** [vikejs/vike#1511] [Discussion] [answered]
Hi, thank you for your library.

I was wondering how can use usesData inside a +Layout.tsx. My usecase is i need to include data from DB in the footer that is on every page of a website.

If so,  how?

**Comment by brillout** [comment]
The `useData()` hook also works in the `<Layout>` component.

**Reply by ierehon1905** [reply]
Hi,
But it seems that the useData returns the data from the +data.ts file that is nearest to the page that is being opened. Do i need to add the required data to every +data.ts or i am missing something?

**Reply by ierehon1905** [reply]
Maybe there is a way to use multiple +data.ts hooks on a single page?

**Comment by ierehon1905** [comment, marked as answer]
Thank you @brillout i found a solution in the documentation. (I should have read it more thourouly). It clearly states there what to do in my case :)
https://vike.dev/onBeforeRender#override
> Using [data() hooks](https://vike.dev/data): you can then use one global onBeforeRender() while using page-specific data() hooks.

**Reply by brillout** [reply]
👍 Btw. we're looking for [sponsors](https://github.com/vikejs/vike/issues/1350), in case that's something your company would be up for.

**Reply by ierehon1905** [reply]
It for my side project, but I enjoy working with vike so far a lot. So I would consider it. Again thank you.
